app/uk/gov/hmrc/agentmappingfrontend/controllers/SignedOutController.scala

```diff
@@ -33,8 +34,27 @@ class SignedOutController @Inject()(appConfig: AppConfig) extends MappingBaseCon
     Redirect(signOutAndRedirectUrl)
   }
 
+  def taskListSignOutAndRedirect(id: MappingArnResultId): Action[AnyContent] = Action { implicit request =>
+    val url = s"${appConfig.taskListSignOutRedirectUrl}?id=$id"
+    Redirect(constructRedirectUrl(url))
+  }
+
+  private def constructRedirectUrl(continue: String): String =
+    s"${appConfig.companyAuthFrontendExternalUrl}${appConfig.ggSignIn}?continue=${urlEncode(continue)}"
+
   def reLogForMappingStart: Action[AnyContent] = Action { implicit request =>
     Redirect(appConfig.signInAndContinue).withNewSession
   }
 
+  def taskList(): Action[AnyContent] = Action.async { implicit request =>
+    val url = s"${appConfig.agentSubscriptionFrontendExternalUrl}${appConfig.agentSubscriptionFrontendTaskListPath}"
+    Future.successful(Redirect(url))
+  }
+
+  def returnAfterMapping(): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
don't think we need this method can just construct the url in AppConfig and then call Redirect on it, also the return url takes a query param id which is the continueId for the record